Sources/PointFree/Account/Account.swift

```diff
@@ -426,61 +426,86 @@ public func nextBilling(for subscription: Stripe.Subscription) -> String {
+    gridColumn(sizes: [.mobile: 3], [
+      p([div(["Status"])])
+      ]),
+    gridColumn(sizes: [.mobile: 9], [
+      gridRow([
+        gridColumn(sizes: [.mobile: 12, .desktop: 6], [
+          div([`class`([Class.padding([.mobile: [.leftRight: 1]])])], [
+            p([text(status(for: subscription))])
+            ])
+          ])
+        ])
+      ])
+    ])
+
+  let nextBillingRow = subscription.cancelAtPeriodEnd || subscription.status == .canceled
+    ? nil
+    : gridRow([
       gridColumn(sizes: [.mobile: 3], [
-        p([div(["Plan"])])
+        p([div(["Next billing"])])
         ]),
       gridColumn(sizes: [.mobile: 9], [
-        gridRow([
-          gridColumn(sizes: [.mobile: 12, .desktop: 6], [
-            div([`class`([Class.padding([.mobile: [.leftRight: 1]])])], [
-              p([text(planName(for: subscription))])
-              ])
-            ]),
-          gridColumn(sizes: [.mobile: 12, .desktop: 6], [
-            div([`class`([Class.padding([.mobile: [.leftRight: 1]]), Class.grid.end(.desktop)])], [
-              p([mainAction(for: subscription)])
-              ])
-            ])
+        div([`class`([Class.padding([.mobile: [.leftRight: 1]])])], [
+          p([text(nextBilling(for: subscription))])
           ])
         ])
-      ]),
+      ])
+
+  let discountRow = subscription.discount.map { discount in
     gridRow([
       gridColumn(sizes: [.mobile: 3], [
-        p([div(["Status"])])
+        p([div(["Discount"])])
         ]),
       gridColumn(sizes: [.mobile: 9], [
-        gridRow([
-          gridColumn(sizes: [.mobile: 12, .desktop: 6], [
-            div([`class`([Class.padding([.mobile: [.leftRight: 1]])])], [
-              p([text(status(for: subscription))])
-              ])
-            ])
+        div([`class`([Class.padding([.mobile: [.leftRight: 1]])])], [
+          p([text(discountDescription(for: discount))])
+          //
           ])
         ])
       ])
-    ]
-    + (
-      subscription.cancelAtPeriodEnd || subscription.status == .canceled
-        ? []
-        : [
-          gridRow([
-            gridColumn(sizes: [.mobile: 3], [
-              p([div(["Next billing"])])
-              ]),
-            gridColumn(sizes: [.mobile: 9], [
-              div([`class`([Class.padding([.mobile: [.leftRight: 1]])])], [
-                p([text(nextBilling(for: subscription))])
-                ])
-              ])
-            ])
-      ]
-    )
+  }
+
+  return div(
+    [`class`([Class.padding([.mobile: [.top: 1, .bottom: 3]])])],
+    [planRow, statusRow] + [nextBillingRow, discountRow].compactMap(id)
   )
 }
 
+private func discountDescription(for discount: Stripe.Subscription.Discount) -> String {
+  var result = "\(discount.coupon.name): "
+  if let percentOff = discount.coupon.percentOff {
+    result += "\(Int(percentOff))% off"
+  } else if let amountOff = discount.coupon.amountOff {
+    result += "$\(amountOff) off"
+  }
+  return result
```

Comment:
example of what this looks like from screenshot test:

<img width="467" alt="image" src="https://user-images.githubusercontent.com/135203/43610044-357e693a-9673-11e8-835c-03cd0c38a13e.png">